codex/slots/converters.nim

```diff
@@ -0,0 +1,89 @@
+## Nim-Codex
+## Copyright (c) 2024 Status Research & Development GmbH
+## Licensed under either of
+##  * Apache License, version 2.0, ([LICENSE-APACHE](LICENSE-APACHE))
+##  * MIT license ([LICENSE-MIT](LICENSE-MIT))
+## at your option.
+## This file may not be copied, modified, or distributed except according to
+## those terms.
+
+import std/sequtils
+
+import pkg/libp2p
+import pkg/stew/arrayops
+import pkg/questionable
+import pkg/questionable/results
+import pkg/poseidon2
+import pkg/poseidon2/io
+
+import ../codextypes
+import ../merkletree
+import ../errors
+import ../utils/digest
+
+func toCid(hash: Poseidon2Hash, mcodec: MultiCodec, cidCodec: MultiCodec): ?!Cid =
+  let
+    mhash = ? MultiHash.init(mcodec, hash.toBytes).mapFailure
+    treeCid = ? Cid.init(CIDv1, cidCodec, mhash).mapFailure
+  success treeCid
+
+proc toPoseidon2Hash(cid: Cid, mcodec: MultiCodec, cidCodec: MultiCodec): ?!Poseidon2Hash =
+  if cid.cidver != CIDv1:
+    return failure("Unexpected CID version")
```

Comment:
This check already happens for all `CIDs` that aren't `dag-pb`, but I guess it doesn't hurt.